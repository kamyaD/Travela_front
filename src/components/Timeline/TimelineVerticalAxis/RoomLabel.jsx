/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import MaintainceForm from '../../Forms/MaintainanceForm';
import tick from '../../../images/Tick/tick.svg';
import Modal from '../../modal/Modal';

class RoomLabel extends PureComponent {
  state = {
    showMarkUnavailable: false
  };

  toggleMarkUnavailable = () => {
    this.setState(prevState => ({
      showMarkUnavailable: !prevState.showMarkUnavailable
    }));
  }

  renderMainteinanceForm(type){
    const {closeModal, name, id, status, shouldOpen, addmaintenanceRecord,
      updateMaintenanceRecord, modalType, timelineDateRange, guestHouseId} = this.props;
    return(
      <Modal
        closeModal={closeModal}
        width="480px"
        customModalStyles="room-maintanance"
        visibility={shouldOpen && type === modalType ? 'visible' : 'invisible'}
        title={`Mark ${name} Unavailable`}
      >
        <MaintainceForm
          status={status}
          id={id}
          closeModal={closeModal}
          addmaintenanceRecord={addmaintenanceRecord}
          timelineDateRange={timelineDateRange}
          guestHouseId={guestHouseId}
          updateMaintenanceRecord={updateMaintenanceRecord}
        />
      </Modal>
    );
  }

  render() {
    const {showMarkUnavailable} = this.state;
    const {name, id, openModal, status} = this.props;
    const visibility = showMarkUnavailable ? 'is-visible' : 'is-hidden';
    const statusClass = status ? 'fault' : 'fine';
    return (
      <div className="room-name item-row">
        <div className="name">{name}</div>
        <div
          className="ellipsis"
          tabIndex="0"
          role="button"
          onClick={this.toggleMarkUnavailable}
        >
          &hellip;
          <div 
            className={`mark-unavailable ${visibility}`}
            role="button"
            tabIndex="0"
            onClick={() => !status && openModal(true,  `${name}-${id}`)}
          >
            <div className={`container_room_${statusClass}`} />
            <span>Unavailable</span>
          </div>
        </div>
        {this.renderMainteinanceForm(`${name}-${id}`)}
      </div>
    );
  }
}

RoomLabel.propTypes = {
  closeModal: PropTypes.func.isRequired,
  addmaintenanceRecord: PropTypes.func.isRequired,
  updateMaintenanceRecord: PropTypes.func.isRequired,
  shouldOpen: PropTypes.bool.isRequired,
  status: PropTypes.bool.isRequired,
  timelineDateRange: PropTypes.array.isRequired,
  guestHouseId: PropTypes.string.isRequired,
  modalType: PropTypes.string,
  openModal: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
RoomLabel.defaultProps = {
  modalType: null
};

export default RoomLabel;
