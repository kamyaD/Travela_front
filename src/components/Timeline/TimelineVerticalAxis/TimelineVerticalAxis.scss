.vertical-axis {
  width: 125px;
  border-right: 1px solid #E4E4E4;
  padding-top: 81px;
  background-color: #FFFFFF;
  position: sticky;
  left: 0;
  z-index: 20;
  box-sizing: border-box;
  display: flex;
  flex-flow: column;
  box-sizing: border-box;

  .room-axis-wrapper {
    padding-left: 15px;
    background-color: #FFFFFF;
    width: 100%;
    box-sizing: border-box;

    &:not(:first-child) {
      border-top: 1px solid #E4E4E4;
    }
    &:last-child {
      padding-bottom: 30px;
    }

    .room-name, .bed-name {
      box-sizing: border-box;
      width: 100px;
    }

    .room-name {
      color: #4F4F4F;
      font-size: 15px;
      font-weight: 500;
      text-transform: uppercase;
      .name{
        overflow: hidden;
      }
      .modal.visible {
        text-transform: none;
      }
      .ellipsis {
        right: 10px;
        text-transform: none;
        margin-left: 15px;
        height: 20px;
        line-height: 20px;
        cursor: pointer;
        transform: translateY(-25%);
        outline: none;
      }
      .mark-unavailable {
        position: absolute;
        left: 60%;
        top: 50%;
        display: flex;
        padding: 0 15px;
        justify-content: space-between;
        align-items: center;
        height: 42px;
        width: 130px;
        border-radius: 2px;
        background-color: #FFFFFF;
        box-shadow: 0 2px 5px 0 rgba(0,0,0,0.15);
        transition: transform .2s ease-out;
        transform-origin: top left;
        outline:none !important;

        &.is-visible {
          transform: scale(1) translateX(0);
          z-index: 200;
        }
        &.is-hidden {
          transform: scale(0) translateX(-100%);
        }
      }
    }
    .bed-name {
      color: #999999;
      font-size: 15px;
      position: relative;
      text-transform: capitalize;
    }
  }
}

.container_room_fault {
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  outline:none !important;
  background: cornflowerblue;
}
.container_room_fine {
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  outline:none !important;
}
.room-maintanance {
  margin-top: 154px;
}

