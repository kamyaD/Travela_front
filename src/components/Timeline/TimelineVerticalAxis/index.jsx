import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import RoomLabel from './RoomLabel';
import './TimelineVerticalAxis.scss';



class TimelineVerticalAxis extends PureComponent {
  render() {
    const {rooms, updateRoomState, timelineDateRange, guestHouseId, updateMaintenanceRecord,
      closeModal, openModal, shouldOpen, addmaintenanceRecord, modalType} = this.props;
    const roomAxisLabels = rooms.map(room => (
      <div key={room.id} className="room-axis-wrapper">
        <RoomLabel
          modalType={modalType}
          shouldOpen={shouldOpen}
          openModal={openModal}
          closeModal={closeModal}
          name={room.roomName}
          id={room.id}
          status={room.faulty}
          updateRoomState={updateRoomState}
          timelineDateRange={timelineDateRange}
          guestHouseId={guestHouseId}
          addmaintenanceRecord={addmaintenanceRecord}
          updateMaintenanceRecord={updateMaintenanceRecord}
        />
        {room.beds.map(bed => (
          <div key={bed.id} className="bed-name item-row">
            <div>{bed.bedName}</div>
          </div>
        ))}
      </div>
    ));

    return (
      <div className="vertical-axis">
        {roomAxisLabels}
      </div>
    );
  }
}

TimelineVerticalAxis.propTypes = {
  rooms: PropTypes.array.isRequired,
  openModal: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  updateRoomState: PropTypes.func.isRequired,
  timelineDateRange: PropTypes.array.isRequired,
  guestHouseId: PropTypes.string.isRequired,
  shouldOpen: PropTypes.bool.isRequired,
  addmaintenanceRecord: PropTypes.func.isRequired,
  updateMaintenanceRecord: PropTypes.func.isRequired,
  modalType: PropTypes.string
};
TimelineVerticalAxis.defaultProps = {
  modalType: null
};

export default TimelineVerticalAxis;
