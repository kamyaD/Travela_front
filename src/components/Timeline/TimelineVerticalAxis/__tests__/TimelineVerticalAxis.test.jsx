import React from 'react';
import TimelineVerticalAxis from '..';

const props = {
  rooms: [
    {
      roomName: 'Ndovu',
      id: 'room-id-1',
      faulty: false,
      beds: [
        {
          bedName: 'Kitanda 1',
          bedId: 'Kitanda1'
        }
      ]
    }
  ],
  updateRoomState: jest.fn(),
  timelineDateRange: ['2018-01-01', '2018-01-31'],
  guestHouseId: 'guest-house-id-1'
};

describe('<TimelineVerticalAxis />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TimelineVerticalAxis {...props} />
    );
  });

  it('renders properly without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
