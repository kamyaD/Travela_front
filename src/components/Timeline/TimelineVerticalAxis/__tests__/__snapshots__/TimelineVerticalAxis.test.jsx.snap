// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineVerticalAxis /> renders properly without crashing 1`] = `
<div
  className="vertical-axis"
>
  <div
    className="room-axis-wrapper"
    key="room-id-1"
  >
    <RoomLabel
      guestHouseId="guest-house-id-1"
      id="room-id-1"
      name="Ndovu"
      status={false}
      timelineDateRange={
        Array [
          "2018-01-01",
          "2018-01-31",
        ]
      }
      updateRoomState={[Function]}
    />
    <div
      className="bed-name item-row"
    >
      <div>
        Kitanda 1
      </div>
    </div>
  </div>
</div>
`;
